hs_app_timeseries/models.py

```diff
@@ -950,211 +1106,462 @@ def delete_all_elements(self):
-                    self._update_actionby_table_insert(con, cur, people_data)
-
-                    # since we are allowing user to set the UTC offset in case of CSV file
-                    # upload we have to update the actions table
-                    self._update_utcoffset_related_tables(con, cur)
-
-                    # update resource specific metadata
-                    self._update_variables_table(con, cur)
-                    self._update_methods_table(con, cur)
-                    self._update_processinglevels_table(con, cur)
-                    self._update_sites_related_tables(con, cur)
-                    self._update_results_related_tables(con, cur)
-
-                    # update CV terms related tables
-                    self._update_CV_tables(con, cur)
-
-                    # push the updated sqlite file to iRODS
-                    utils.replace_resource_file_on_irods(temp_sqlite_file, sqlite_file_to_update,
-                                                         user)
-                    self.is_dirty = False
-                    self.save()
-                    log.info("SQLite file update was successful.")
-            except sqlite3.Error as ex:
-                sqlite_err_msg = str(ex.args[0])
-                log.error("Failed to update SQLite file. Error:{}".format(sqlite_err_msg))
-                raise Exception(sqlite_err_msg)
-            except Exception as ex:
-                log.exception("Failed to update SQLite file. Error:{}".format(ex.message))
-                raise ex
-            finally:
-                if os.path.exists(temp_sqlite_file):
-                    shutil.rmtree(os.path.dirname(temp_sqlite_file))
+    def update_datasets_table_insert(self, con, cur):
+        # insert record to Datasets table - first delete any existing records
+        # used for updating a sqlite file that is blank (case of CSV upload)
 
-    def populate_blank_sqlite_file(self, temp_sqlite_file, user):
-        """
-        writes data to a blank sqlite file. This function is executed only in case
-        of CSV file upload and executed only once when updating the sqlite file for the first time.
-        :param temp_sqlite_file: this the sqlite file copied from irods to a temp location on
-         Django.
-        :param user: current user (must have edit permission on resource) who is updating the
-        sqlite file to sync metadata changes in django database.
-        :return:
-        """
-        log = logging.getLogger()
-        if not self.resource.has_csv_file:
-            msg = "Resource needs to have a CSV file before a blank SQLite file can be updated"
-            log.exception(msg)
-            raise Exception(msg)
+        cur.execute("DELETE FROM Datasets")
+        con.commit()
+        insert_sql = "INSERT INTO Datasets (DatasetID, DatasetUUID, DatasetTypeCV, " \
+                     "DatasetCode, DatasetTitle, DatasetAbstract) VALUES(?,?,?,?,?,?)"
```

Comment:
Wondering whether there should be three ```?``` in the insert sql statement above since there are three variable values to insert?

Reply:
There are 6 columns for which data is getting passed so 6 '?'

Reply:
@pkdash OK. I missed the line above it, yes, now I see 6 columns.